src/Context/SoapContext.php

```diff
@@ -273,4 +273,15 @@ public function beforeStepCheckForException(BeforeStepScope $scope)
             throw $this->fault;
         }
     }
+
+    /**
+     * @Given I register the following headers:
+     */
+    public function iRegisterHeaders(TableNode $headers)
+    {
+        foreach ($headers->getRowsHash() as $key => $value) {
+            $soapHeaders[] = new \SoapHeader($key, $key, $value);
```

Comment:
@asgorobets Thank you for your feedback. I have made an update to support the namespace. I think that the array of data thing can come as a separate feature if people ask for it . Wdyt ?